scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -1307,17 +1317,37 @@ def __select_processes(self):
                 self.__pids = [self.__pids[0]]
         else:
             # See if the specified target pid is running.  If so, then return it.
-            # Special case '$$' to mean this process.
+            # Special cases:
+            #   '$$' mean this process.
+            #   '$$TBD' mean that the PID of the target process has not been determined yet and it will be set later.
             pids = []
             if self.__target_pids:
                 for t_pid in self.__target_pids:
                     if t_pid == "$$":
                         t_pid = int(os.getpid())
+
+                    # skip this until it will be replaced with a real PID.
+                    elif t_pid == "$$TBD":
+                        continue
                     else:
                         t_pid = int(t_pid)
                     pids.append(t_pid)
             self.__pids = pids
         return self.__pids
 
+    def set_pid(self, pid):  # type: (int) -> None
+        """
+        Set the PID of the process that was marked as $$TBD.
+        :param pid: Process PID
+        """
+        for i in range(len(self.__target_pids)):
```

Comment:
To confirm I'm understanding this correctly - there can only be a single entry with ``$$TBD`` in this list since we have this class instance per monitor, right?

Reply:
Right, the meaning is the same as for the agent process monitor with "$$" - we need to track only one particular process.